smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/OutgoingKafkaRecord.java

```diff
@@ -37,7 +40,8 @@ public OutgoingKafkaRecord(String topic, K key, T value, Instant timestamp, int
                 .orElseThrow(() -> ex.illegalArgumentNoMetadata(OutgoingKafkaRecordMetadata.class));
```

Comment:
That needs to be changed too. You can't expect to have OutgoingKafkaRecordMetadata, also you need to be sure you keep the existing set. To do that, the `OutgoingKafkaRecord ` constructor should receive the original Metadata.

Reply:
I tried retaining the incoming metadata, but managed to break other tests! I can try again though.

What would be a good way to test that the current approach is losing metadata?